### DISCUSSION TITLE
How to implement Start and Stop Transaction functionalities via external application in steVe OCPP?

### DISCUSSION THREAD

**Discussion by rom232** [steve-community/steve#1624] [Q&A] [answered]
Hello, steVe OCPP community!

I am currently working on a project that requires implementing the functionalities for Start Transaction, Stop Transaction, Start Remote Transaction, and Stop Remote Transaction from an external application. I would like to know if there is support for making these calls through a REST API, WebSocket, or any other protocol such as SOAP.

Is there a documented REST API that I can use to send commands for Start Transaction, Stop Transaction, Start Remote Transaction, and Stop Remote Transaction?
If there is no REST API available, is it possible to implement these functionalities using WebSocket? If so, what are the necessary steps to connect and interact with the server?
If there are any recommended approaches for integrating my external application with steVe OCPP, I would appreciate any guidance or examples you can provide.
Thank you in advance for your help, and I look forward to learning more about integrating with steVe OCPP!

**Comment by juherr** [comment, marked as answer]
You'll have to implement the missing apis first.
See #1000

**Reply by rom232** [reply]
Thank you for the response!

My colleagues are trying to make station management calls directly via SOAP from an external source (an app or backend server). But I understand that steVe does not currently provide this type of management, am I correct?

Do you know if and how anyone has managed to perform these operations without significant changes to the codebase? I saw that #1000 has not been accepted yet, so the rest of the API would have to be implemented locally, correct?

Thank you.

**Reply by juherr** [reply]
Steve provides a full station management from the UI.
A pull request is already available for remote start/stop: https://github.com/steve-community/steve/pull/1291

**Reply by rom232** [reply]
Hi, sorry to bother you again.

I'm trying to use the start/end API endpoint, but I'm receiving the following error:

```json
{
    "timestamp": "2024-11-11T16:18:06.662Z",
    "status": 500,
    "error": "Internal Server Error",
    "message": "Cannot invoke \"java.lang.Integer.intValue()\" because \"connectorId\" is null",
    "path": "https://ocpp.7rk.me/steve/api/v1/remote/start"
}



My request:

curl --location 'https://ocpp.<domain>/api/v1/remote/start' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic YWRtaW46MTIzNA==' \
--data '{
    "chargeBoxId": "JDBK4300040AF",
    "connectorId": 1,
    "idTag": "739E7EDF"
}'


thanks

**Reply by Shivk14** [reply]
@rom232 we can call remote start and stop transaction from external application through RESTApi which u need to implement not implement exactly already steve have the service just you need to expose through REST we can call. I did same thing it is working.

**Comment by goekay** [comment]
https://github.com/steve-community/steve/pull/1949 was merged, providing API implementation for all OCPP operations.
